docs/src/pages/demos/buttons/ToggleButtonsControlledState.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from 'material-ui/styles';
+import FormatAlignLeft from 'material-ui-icons/FormatAlignLeft';
+import FormatAlignCenter from 'material-ui-icons/FormatAlignCenter';
+import FormatAlignRight from 'material-ui-icons/FormatAlignRight';
+import FormatAlignJustify from 'material-ui-icons/FormatAlignJustify';
+import FormatBold from 'material-ui-icons/FormatBold';
+import FormatItalic from 'material-ui-icons/FormatItalic';
+import FormatUnderlined from 'material-ui-icons/FormatUnderlined';
+import FormatColorFill from 'material-ui-icons/FormatColorFill';
+import ArrowDropDownIcon from 'material-ui-icons/ArrowDropDown';
+import Typography from 'material-ui/Typography';
+import ToggleButton, { ToggleButtonGroup } from 'material-ui/ToggleButton';
+
+const styles = theme => ({
+  toggleContainer: {
+    height: 56,
+    padding: `${ theme.spacing.unit }px ${ theme.spacing.unit * 2 }px`,
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'flex-start',
+    margin: `${ theme.spacing.unit }px 0`,
+    background: theme.palette.background.default,
+  },
+});
+
+function doSomething(event) {
+  // eslint-disable-next-line no-console
+  console.log(event.currentTarget.getAttribute('data-something'));
+}
+
+function ToggleButtonWithValue({ value, values, onClick, ...other }) {
+  return (
+    <ToggleButton
+      { ...other }
+      onClick={ e => onClick( e, value ) }
+      selected={ values.indexOf( value ) >= 0 }
+    />
+  );
+}
+
+class ToggleButtonsControlledState extends React.Component {
+  state = {
+    alignment: [],
+    formats: [],
+  };
+
+  handleToggleFormat = ( e, value ) => {
+    const { formats } = this.state;
+    const index = formats.indexOf(value);
+
+    if (index === -1) {
+      this.setState({ formats: [ ...formats, value ] });
+    } else {
+      formats.splice(index, 1);
+      this.setState({ formats });
+    }
+  }
+
+  handleToggleAlignment = ( e, value ) => {
+    const { alignment } = this.state;
+
+    if (alignment.length === 0 || alignment[0] !== value) {
+      this.setState({ alignment: [ value ] });
+    } else {
+      this.setState({ alignment: [] });
+    }
+  }
+
+
+  render() {
+    const { classes } = this.props;
+    const { alignment, formats } = this.state;
+
+    return (
+      <div className={ classes.toggleContainer }>
+        <ToggleButtonGroup selected={ alignment.length > 0 }>
+          <ToggleButtonWithValue value="left" values={ alignment } onClick={ this.handleToggleAlignment }><FormatAlignLeft /></ToggleButtonWithValue>
```

Comment:
jsx attributes shouldn't be spaced: `prop={value}` I don't think it'll pass `npx lint` like so (try to have your editor format/eslint --fix on save)

Reply:
Thanks for the note. This is currently WIP. I'll be sure to lint everything when it's totally ready. (normally have editor do it automatically but haven't set it up for this project).